docs/DeveloperGuide.md

```diff
@@ -8,8 +8,237 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### EasInternship (main class)
 
-## Product scope
+1. [Design](#design)
+   1. [Architecture](#architecture)
+   2. [UI Component](#ui-component)
+   3. [Logic Component](#logic-component)
+   4. [Model Component](#model-component)
+   5. [Storage Component](#storage-component)
+2. [Implementation](#implementation)
+   1. [Parsing Input](#parsing-input)
+   2. [Command Execution](#command-execution)
+   3. [Task Management](#task-management) 
+   4. [AddCommand Implementation](#addcommand-implementation)
+   5. [DeleteCommand Implementation](#deletecommand-implementation)
+   6. [SortCommand Implementation](#sortcommand-implementation)
+   7. [UpdateCommand Implementation](#updatecommand-implementation)
+
+---
+
+## 1. Design
+
+### 1.1 Architecture
+The architecture of EasInternship is designed to follow the MVC (Model-View-Controller) pattern to facilitate separation of concerns, modularity, and maintainability.
+
+- **View (UI)**: Responsible for interacting with the user by displaying output and reading input.
+- **Controller (Command and Parser)**: Responsible for parsing user input and invoking the appropriate commands.
+- **Model (InternshipList)**: Manages the state of the application, including the list of internships and tasks.
+- **Storage**: Responsible for loading and saving data from and to the disk.
+
+---
+
+### 1.2 UI Component
+
+The UI component is responsible for displaying information to the user and reading input. It interacts with the user by printing messages and prompts, and retrieves commands for the rest of the application to process.
+
+### 1.3 Logic Component
+
+The logic component is handled by the `Parser` and `Command` classes. The `Parser` interprets the user's input and returns a command object, which is then executed to perform the desired operation.
+
+### 1.4 Model Component
+
+The model component includes the `Internship`, `InternshipList`, `Deadline` and `DeadlineList` classes.
+
+`Internship`: Represents the individual internship entries.
+
+`InternshipList`: Manages a collection of Internship objects, providing methods for adding, retrieving, updating, sorting and removing internships.
+
+`Deadline`: Represents each individual deadline associated with an internship, holding data specific to that deadline (description, date).
+
+Each `Internship` object contains a `Deadline` list, and has methods to add, retrieve or remove deadlines associated with that internship.
+
+This is represented with the class diagram below:
+
+![](UML/Internship-Model_Component.png)
```

Comment:
I believe the visibility of the methods and attributes in this diagram should be signs like "-" and "+" instead of squares and circles. You can consider configuring your plantuml to reflect this.